src/kv/mod.rs

```diff
@@ -175,16 +220,26 @@ impl KeyRange {
         if key.is_empty() {
             // An empty Vec<u8> results in an invalid KeyRange.
             // Assume that an empty value passed to this method implies no prefix (i.e., all keys).
-            return KeyRange::all();
+            return Self::all();
         }
 
         let range_end = {
             let mut end = key.clone();
 
             for i in (0..end.len()).rev() {
```

Comment:
this for loop looks kinda strange, can you rewrite it in functional style for easy understanding?

Reply:
Sorry, what do you mean by functional style? I added some comments to make it more readable.

Reply:
Try to use functional style (iterator, map, filter...).